src/EpiEasyEvents/EventsRegistry.cs

```diff
@@ -230,7 +230,7 @@ private void HandleEvent<TEventArgs>(Type handlerInterface, TEventArgs eventArgs
             foreach (var handler in eventHandlers)
             {
                 var handleMethod = handler.GetType()
-                    .GetMethod(nameof(IContentEventHandler<PageData, ContentEventArgs>.Handle));
+                    .GetMethod(nameof(IContentEventHandler<PageData, ContentEventArgs>.Handle), new Type[]{eventArgs.Content.GetType(), eventArgs.GetType()});
```

Comment:
you can make use of new syntax and omit the type. Also, a type of content is already checked a bit above with additional null check, which is lacking in your version. So, these two changes together can make line 233 look like:

`.GetMethod(nameof(IContentEventHandler<PageData, ContentEventArgs>.Handle), new []{ pageType, eventArgs.GetType()});`